### PR TITLE
Add K8s Authenticator Quickstart

### LINKED ISSUE #246: There is a quick start guide for Conjur Kubernetes authentication
Use: 
- Cluster prep Helm chart from #227
- Namespace prep Helm chart from #236
- Sample application deployment Helm chart from Task #238

The goal is to demonstrate setting up an E2E flow for using Kubernetes (in the spirit of https://github.com/cyberark/dap-wiki/tree/master/how-to-guides/kubernetes#set-up-and-use-kubernetes-authenticator). We envisage this as a decomposed guide that uses the reusable scripts in order.

~This could also be adopted into an interactive experience like in Katacoda.~

1. Reproducible steps for building an environment locally (e.g. KinD).
2. Sane defaults.
3. Follow a singular happy path.

The singular happy path can be:
- KinD cluster
- Conjur OSS deployed in cluster
- [DAP wiki flow](https://github.com/cyberark/dap-wiki/tree/master/how-to-guides/kubernetes#set-up-and-use-kubernetes-authenticator) to deploy [Pet store demo app](https://github.com/conjurdemos/pet-store-demo/) with Conjur sidecar

### PR DESCRIPTION
**Pull request by john-odonnell**
### What does this PR do?
Includes `bin/test-workflow/QUICKSTART.md`
Guide uses reusable scripts added in #307 

The quickstart guide will only improve with updates
addressing #322 and #323 - guide will be able to divide
steps more clearly by persona responsability.

### What ticket does this PR close?
Resolves #246 

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

#### Manual tests
**If you are preparing for a release**, have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local authn-k8s client image build

### CODE REVIEW

**Comment by diverdane** [comment]
@john-odonnell  This looks good! This should make it easy to create a Katacoda scenario at some point!

Maybe you can add a note in the Description that the QuickStart will improve when this Issue has been addressed (i.e. we can eliminate the local build of pet store):
https://github.com/cyberark/conjur-authn-k8s-client/issues/318

Another consideration is that in some places, it might be nice to have a sample output, or maybe a snippet of output to look for. (We've been keeping output code blocks separate from the commands to avoid having to add prompts and thereby making it awkward to cut/paste the commands.) Not that we need to go crazy with output for every command, just wherever it is helpful or can boost user confidence.

File: bin/test-workflow/QUICKSTART.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+* Starting a kind cluster with local Docker registry
+* Creating a new namespace for Conjur, `conjur-oss`
+* Deploying Conjur OSS with Helm
+* Enable the Kubernetes Authenticator in Conjur
+
+The Conjur OSS Helm Chart is published on
+[GitHub](https://github.com/cyberark/conjur-oss-helm-chart).
+This step wraps a Conjur OSS kind deployment
+[example](https://github.com/cyberark/conjur-oss-helm-chart/tree/main/examples/kubernetes-in-docker)
+found there.
```
Comment (doodlesbykumbi) [inline bin/test-workflow/QUICKSTART.md:36]: I think it might be useful to rephrase this to provide more detail about the example folder and call out the scripts.

```suggestion
The Conjur OSS Helm Chart repository contains an [example](https://github.com/cyberark/conjur-oss-helm-chart/tree/main/examples/kubernetes-in-docker) folder with scripts and instructions for deploying the Conjur OSS Helm Chart on KinD. The scripts from the example folder are used to accomplish this step by git cloning the Conjur OSS Helm Chart repository and using them to carry out the tasks mentioned above.
```

Comment (doodlesbykumbi) [reply bin/test-workflow/QUICKSTART.md:36]: I think we should also explicitly mention the need for git, can't be too careful.

File: bin/test-workflow/QUICKSTART.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+# Conjur Kubernetes Authentication Quick Start Guide
+
+The following guide details an end-to-end workflow for using Conjur Kubernetes
+authentication by deploying a PetStore demo application to a
+Kubernetes-in-Docker (kind) cluster.
+
+At the end of this quickstart, you will have:
+
+* Deployed Conjur OSS to a kind cluster
```
Comment (doodlesbykumbi) [inline bin/test-workflow/QUICKSTART.md:9]: This is purely an observation comment and you don't need to act on it other than potentially creating an issue for future work. 

It's definitely simpler at this point to be prescriptive in this guide and mention only using the KinD cluster. However, in future work it would be nice if the KinD step became optional so that users could bring their own Kubernetes cluster. The requirement being that `kubectl` is already setup to communicate to their cluster of choice. This would change the steps of the quickstart from:


* Deployed Conjur OSS to a kind cluster
* ...

to:

* Created a KinD cluster (optional)
* Deployed Conjur OSS to a Kubernetes cluster
* ...

**Review by doodlesbykumbi — COMMENTED** [review]
Looking good. Left some preemptive comments

File: bin/test-workflow/QUICKSTART.md (hunk 1 above)
Comment (john-odonnell) [reply bin/test-workflow/QUICKSTART.md]: - Added detail to mention of Conjur OSS Helm Chart example
- Added `git` to list of prerequisites

File: bin/test-workflow/QUICKSTART.md (hunk 2 above)
Comment (john-odonnell) [reply bin/test-workflow/QUICKSTART.md]: Filed #329

**Review by doodlesbykumbi — APPROVED** [review]
Thanks for addressing those comments. LGTM, though I noticed there's no example output for some of the command but hardly a blocker.

**Comment by john-odonnell** [comment]
@doodlesbykumbi I didn't include output for steps 6 and 7, they don't output any logs that succinctly indicate success. I'll file an issue to make them more descriptive in success.
